writer.go

```diff
@@ -172,6 +173,110 @@ loop:
 	return nil
 }
 
+// RespondWithContext reads Events from a channel and writes SSE HTTP response,
+// with context awareness for proper connection handling. This allows for graceful
+// termination when the client disconnects or the request is canceled.
+//
+// Ctx is used to detect when the client disconnects or the request is canceled.
+// W is the response writer for the SSE stream.
+// Source provides the events to be sent to the client.
+// Cfg is SSE stream configuration, if nil is passed configuration from
+// DefaultConfiguration global will be used.
+// Stop is an optional channel for stopping SSE stream, if this channel is
+// closed SSE stream will stop and http connection closed.
+//
+// This function returns nil if end of stream is reached, stream lifetime
+// expired, client closes the connection or request to stop is received on the
+// stop channel. Otherwise, it returns an error.
+func RespondWithContext(ctx context.Context, w http.ResponseWriter, source <-chan *Event, cfg *Config, stop <-chan struct{}) error {
```

Comment:
I wonder if it would be possible to completely remove `stop <-chan struct{}`? It could be completely replaced by `context.Context`

Reply:
As discussed it seems that dropping `stop <-chan struct{}` and adding `context` per stream with its own handling creates way more code without any additional benefit.